### DISCUSSION TITLE
Support for Veikk VK640?

### DISCUSSION THREAD

**Discussion by JayLow3** [OpenTabletDriver/OpenTabletDriver#1509] [General]
There really aren't any Osu tablet drivers out there that support this model as of the moment. :(  So I was just wondering, will the Veikk VK640 ever get supported? The default Veikk driver is dog shit btw.

**Comment by InfinityGhost** [comment]
This absolutely can be done, install OpenTabletDriver and go to <kbd>Help</kbd> -> <kbd>Export Diagnostics</kbd> while your tablet is plugged in and the VEIKK driver is killed. Send that file here (you might have to change the file extension) and we can get working on it.

**Reply by JayLow3** [reply]
[VEIKK VK640.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7356658/VEIKK.VK640.txt)

**Reply by InfinityGhost** [reply]
Update: https://github.com/OpenTabletDriver/OpenTabletDriver/pull/1211 has a configuration for your tablet (I should have searched first before asking for the diagnostics), but there's some stuff that is broken but not vital for basic functionality.

**Reply by JayLow3** [reply]
Ok, so I downloaded the OTD master and transferred the VK640 config to the OTD config folder. Tbh, not sure if that was what I was supposed to do. But when I open OTD it would say that the tablet is detected, but in the output panel, it says no tablet detected.
![SS](https://user-images.githubusercontent.com/92371266/137614462-f59cce66-942f-4815-876c-eea121e4a8e6.png)

Here are the diagnostics if needed.

[VK640.zip](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7358925/VK640.zip)

**Reply by InfinityGhost** [reply]
The master version does not support this configuration format. This displays "no tablet detected" because it was unable to read the dimensions so it assumes all zeroes, however it does actually detect. Use [this](https://github.com/OpenTabletDriver/OpenTabletDriver/actions/runs/1350589421) (download the OpenTabletDriver win-x64 artifact) and it should detect correctly. If you save with this version however, your settings will not be able to transfer back to and older version of the driver.

**Reply by JayLow3** [reply]
Dude it works! Thanks a lot :)

**Reply by BrandsGonza** [reply]
May I also get help I am struggling to make this work. Thank you
[diag for tab.txt](https://github.com/OpenTabletDriver/OpenTabletDriver/files/7796777/diag.for.tab.txt)

**Reply by jamesbt365** [reply]
Release version of otd and this:
https://mrcubix.github.io/OTD.Backport-Wiki/installation.html
